src/components/postgresConnect.ts

```diff
@@ -376,22 +377,25 @@ class PostgresDB {
     }
   }
 
-  public getUserDataByUsername = async (username: string) => {
-    const text = `
-      SELECT id,
-      first_name as "firstName",
-      last_name as "lastName",
-      id as "userId",
-      username, date_of_birth as "dateOfBirth",
-      email,
-      password_hash as "passwordHash"
-      FROM "user"
-      WHERE username = $1
-    `
+  public getUserOrgData = async ({ username, userId }: any) => {
```

Comment:
Since this PR has now gone through quite a few iterations, I am happy to have it this way (with username and userID), so not a required change

Ideally though:

* Keep DB abstraction free from logic (i.e. methods getUserOrgDataByUserId, getuserOrgDataByUsername)
* The view has casting to camelCase alias (i.e. just `SELECT *` when view is used)

Reply:
>     * Keep DB abstraction free from logic (i.e. methods getUserOrgDataByUserId, getuserOrgDataByUsername)

Right, just that the two methods would be basically identical except for the WHERE criteria.

>     * The view has casting to camelCase alias (i.e. just `SELECT *` when view is used)

The thing I don't like about that approach is that you have to remember to quote the camelCase fields everytime you refer to them:
```WHERE "userId" = $1```

Isn't it better to leave everything in postGres as underscore_case and just do the conversions outside?

Reply:
Hmm, I can this subject having mixed opinions. I should have worded my comment better, instead stead of `Ideally though` -> `In my opinion`. 

> Right, just that the two methods would be basically identical except for the WHERE criteria.

Just FYI I've done something similar in the permissions PR:

https://github.com/openmsupply/application-manager-server/blob/af2881923936ee76a0b905c6fae9fa1e4de5e9e7/database/buildSchema/13_template_permission.sql#L11-L29

https://github.com/openmsupply/application-manager-server/blob/af2881923936ee76a0b905c6fae9fa1e4de5e9e7/src/components/postgresConnect.ts#L533-L534

https://github.com/openmsupply/application-manager-server/blob/af2881923936ee76a0b905c6fae9fa1e4de5e9e7/src/components/postgresConnect.ts#L544-L545

(I just realised it's kind of weird that the view is names with kebab but the output columns are all camel, I think I should have named the view with camel, which would have helped to indicate that it's fields are also in camel)

>Isn't it better to leave everything in postGres as underscore_case and just do the conversions outside?

This kind of brings on that `case` discussion. A per our previous convo, I am still of the opinion that everything should just be one type of case (even if it requires quotation marks). But I guess I won't be pushing this point forward. 

This PR looks good, just pending test as per https://github.com/openmsupply/application-manager-server/pull/180#issuecomment-761894339